src/V3Config.cpp

```diff
@@ -145,9 +146,310 @@ class V3ConfigIgnores {
+            }
+        }
+    }
+};
+
+V3ConfigModule V3ConfigModule::s_singleton;
+
+//######################################################################
+// function/task inline, public
+
+class V3ConfigFTask {
+    typedef enum { INLINE, PUBLIC } ConfigType;
+    typedef std::vector<std::pair<string, ConfigType> > FTaskList;
+
+    std::map<string, FTaskList> m_configs;
+
+    static V3ConfigFTask s_singleton;
+    V3ConfigFTask() {}
+    ~V3ConfigFTask() {}
+public:
+    inline static V3ConfigFTask& singleton() { return s_singleton; }
+
+    void addFTaskInline(const string& module, const string& task) {
+        m_configs[module].push_back(make_pair(task, INLINE));
+    }
+    void addFTaskPublic(const string& module, const string& task) {
+        m_configs[module].push_back(make_pair(task, PUBLIC));
+    }
+
+    void applyFTask(const AstNodeModule* modulep, AstNodeFTask* ftaskp) {
+        const string& module = modulep->name();
+        const string& ftask = ftaskp->name();
+        std::map<string, FTaskList>::const_iterator mit;
+        for (mit = m_configs.begin(); mit != m_configs.end(); ++mit) {
+            if (VString::wildmatch(module.c_str(), mit->first.c_str())) {
+                FTaskList::const_iterator fit;
+                for (fit = mit->second.begin(); fit != mit->second.end(); ++fit) {
+                    if (VString::wildmatch(ftask.c_str(), fit->first.c_str())) {
+                        AstPragmaType type;
+                        if (fit->second == INLINE) type = AstPragmaType::NO_INLINE_TASK;
+                        else if (fit->second == PUBLIC) type = AstPragmaType::PUBLIC_TASK;
+                        ftaskp->addStmtsp(new AstPragma(ftaskp->fileline(), type));
+                    }
+                }
+            }
+        }
+    }
+};
+
+V3ConfigFTask V3ConfigFTask::s_singleton;
+
+//######################################################################
+// full/parallel case
+
+class V3ConfigCase {
+    typedef std::map<string, std::unordered_set<int>> CaseList;
+
+    CaseList m_parallels;
+    CaseList m_fulls;
+
+    static V3ConfigCase s_singleton;
+    V3ConfigCase() {}
+    ~V3ConfigCase() {}
+public:
+    inline static V3ConfigCase& singleton() { return s_singleton; }
+
+    void addCaseFull(const string& file, int line) {
+        m_fulls[file].insert(line);
+    }
+    void addCaseParallel(const string& file, int line) {
+        m_parallels[file].insert(line);
+    }
+
+    void applyCase(AstCase* nodep) {
+        const string& file = nodep->fileline()->filename();
+        int lineno = nodep->fileline()->lineno();
+
+        for (CaseList::const_iterator it = m_parallels.begin();
+             it != m_parallels.end(); ++it) {
+            if (VString::wildmatch(file.c_str(), it->first.c_str())) {
+                if ((m_parallels[file].find(0) != m_parallels[file].end()) ||
+                    (m_parallels[file].find(lineno) != m_parallels[file].end())) {
```

Comment:
Perhaps should have standard class to save file/line and match them?

In general seems a lot if code for each pragma.  Might be fine but see if that's the best we want to do.

Nit || goes first on line.

Reply:
Yeah, I believe there is some potential to consolidate, especially once the code for caching is added.